CRM/Core/ClassLoader.php

```diff
@@ -103,9 +103,11 @@ public function register($prepend = FALSE) {
       return;
     }
     $civicrm_base_path = dirname(dirname(__DIR__));
+    $composer_autoload = "$civicrm_base_path/vendor/autoload.php";
 
-    // @todo: need some way to check if we should register vendor or not
-    //require_once dirname(dirname(__DIR__)) . '/vendor/autoload.php';
+    if (file_exists($composer_autoload)) {
```

Comment:
Hm. So, the code should probably be changed to look first for the local vendor directory first (will be present in a non-composer based install), but if it's not there, then look for a top-level vendor directory (will be present in a composer-based install).

I'll take a look! I first need to figure out a good way to test (trying `bin/cli.php` seems like it'd be the easiest?) and see the failure when using a Civi-first entry point, and then I can try that code change and see if it fixes it.

Thanks for the info :-)